### PR TITLE
Clear tech debt by migrating remaining uses of application offer tags to use offer uuids.

### PR DESCRIPTION
**Pull request by wallyworld**
Application offer tags were updated to use the uuid not the name. However, for compatibility reasons, some places still relied on the name, specifically the remote entity token lookups used to manage cross model relation exchanges. So the codebase had compatibility shims in various places.

This PR updates the code to only use offer uuids. For dealing with existing data, it needs an upgrade step and model migration enhancements to convert tags in the database to the correct format.

## QA steps

There's 3 scenarios: new cmr deployment, upgrade existing cmr deployment, migrate cmr model.

cmr deployment can be done by: 
deploy `juju-qa-dummy-source` and offering `dummy-source:sink`
in another model, deploy `juju-qa-dummy-sink` and relate to the dummy-source offer
in the dummy-source app, set `token` config and status will show all green

for upgrade test, do the cmr deployment but don't set token
upgrade controller
set token and ensure status shows green on both models

for migrate test, do the cmr deployment but don't set token
migrate the offering model to a new controller
set the token
both models should have green status

after upgrade/migration, look at the content of the `remoteEntities` collection and see records with `applicationoffer-<uuid>`

## Links

**Jira card:** JUJU-5204

### CODE REVIEW

File: state/migration_import.go [hunk 1]
```diff
@@ -1814,12 +1814,20 @@ func (i *importer) remoteEntities() error {
 		src: i.model,
 		dst: i.st.db(),
 	}
+	offerUUIDByName := make(map[string]string)
+	for _, app := range i.model.Applications() {
+		for _, offer := range app.Offers() {
+			offerUUIDByName[offer.OfferName()] = offer.OfferUUID()
```
Comment (hpidcock) [inline state/migration_import.go:1820]: Probably not a problem, as I don't fully understand the logic here, but is it possible to have multiple offer names with different UUIDs?

**Review by hpidcock — APPROVED** [review]
Awesome, thank-you.

File: state/migration_import.go (hunk 1 above)
Comment (wallyworld) [reply state/migration_import.go:1820]: No, a uuid is created for each offer and offer names are unique to a model.

**Pull request merged**